### PR TITLE
GH-352 Expose OPENSSL_INFO and related functions, clarify version and library information functions

### PR DESCRIPTION
**Pull request by h-vn**
OpenSSL 3.0.0 has a number of new version and library information related functions and constants. The existing related functions also need an update.

New information is added to help determining if the library is OpenSSL and LibreSSL

### CODE REVIEW

**Comment by h-vn** [comment]
Note: when merging, do ***not*** squash the commits this time.

File: lib/Net/SSLeay.pod [hunk 1]
```diff
@@ -995,52 +995,101 @@ The following new functions behave in perlish way:
...
+before. SSLeay() was made an alias of OpenSSL_version_num() in OpenSSL
+1.1.0 and LibreSSL 2.7.0.
+
+B<COMPATIBILITY:> OpenSSL_version_num() requires at least
+Net-SSLeay-1.82 with OpenSSL 1.1.0, or Net-SSLeay-1.88 with LibreSSL
+2.7.0.
 
-Gives version number (numeric) of underlaying openssl library.
+Both functions return OPENSSL_VERSION_NUMBER constant (numeric) as
+defined by the underlaying OpenSSL or LibreSSL library.
```
Comment (chrisnovakovic) [inline lib/Net/SSLeay.pod:1013]: nit: `underlaying` -> `underlying`

File: lib/Net/SSLeay.pod [hunk 2]
```diff
@@ -995,52 +995,101 @@ The following new functions behave in perlish way:
...
+
+ LibreSSL returns 0x20000000 always:
+
+ # 0x20000000 => libressl-2.2.1
+ # ...
+ # 0x20000000 => libressl-3.4.2
 
-You can use it like this:
+You can use the version number like this when you know that the
+underlaying library is OpenSSL:
```
Comment (chrisnovakovic) [inline lib/Net/SSLeay.pod:1065]: nit: `underlaying` -> `underlying`

**Review by chrisnovakovic — APPROVED** [review]

File: lib/Net/SSLeay.pod (hunk 1 above)
Comment (h-vn) [reply lib/Net/SSLeay.pod]: Fixed.

File: lib/Net/SSLeay.pod (hunk 2 above)
Comment (h-vn) [reply lib/Net/SSLeay.pod]: Fixed. Also fixed another instance elsewhere on the page.

**Pull request merged**